File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        String d = ""+x ;
        for(int i = 0 ; i < d.length()/2 ; i++ ){          
            if( d.charAt(i) != d.charAt( d.length() -i - 1 )){
                return false;
            }   
        }
        return true;
    }
}

File: 0009-palindrome-number/0009-palindrome-number.cpp
#include <string>

class Solution {
public:
    bool isPalindrome(int x) {
        std::string d = std::to_string(x);  // Convert integer to string
        int y = d.size();
        
        for(int i = 0; i < y / 2; i++) {
            if(d[i] != d[y - i - 1]) {
                return false;
            }
        }
        
        return true;
    }
};
